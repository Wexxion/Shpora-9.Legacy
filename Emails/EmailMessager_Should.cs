using System;
using ApprovalTests.Combinations;
using ApprovalTests.Reporters;
using NUnit.Framework;
using StatePrinting;

namespace Emails
{
    [UseReporter(typeof(DiffReporter))]
    [TestFixture]
    public class EmailMessager_Should
    {
        [Test]
        public void ApproveMessager()
        {
            var st = new Stateprinter();
            st.Configuration.Add(new SbValueConverter());
            var messager = new NewRateEmailMessager();
            CombinationApprovals.VerifyAllCombinations(
                (name, aType, rate) => messager.CreateMessage(name, aType, rate), x => st.PrintObject(x),
                new[] {"Vladimir Putin"},
                new[] {AccountType.Cheque, AccountType.Credit, AccountType.Savings},
                new[] {0.05m, 2, 10, 100, 150, -100});
        }

        [Test]
        public void ApproveMessagerConfig1()
        {
            var st = new Stateprinter();
            st.Configuration.Add(new SbValueConverter());
            var messager = new NewRateEmailMessager();

            Config.Local.IncreaseRate = true;
            Config.Local.IncreaseRateFactor = 1.5m;

            CombinationApprovals.VerifyAllCombinations(
                (name, aType, rate) => messager.CreateMessage(name, aType, rate), x => st.PrintObject(x),
                new[] { "Vladimir Putin" },
                new[] { AccountType.Cheque, AccountType.Credit, AccountType.Savings },
                new[] { 1, 2, 10, 100, 150, -100 });
        }

        [Test]
        public void ApproveMessagerConfig2()
        {
            var st = new Stateprinter();
            st.Configuration.Add(new SbValueConverter());
            var messager = new NewRateEmailMessager();

            Config.Local.IncreaseRate = true;
            Config.Local.IncreaseRateFactor = 0.75m;

            CombinationApprovals.VerifyAllCombinations(
                (name, aType, rate) => messager.CreateMessage(name, aType, rate), x => st.PrintObject(x),
                new[] { "Vladimir Putin" },
                new[] { AccountType.Cheque, AccountType.Credit, AccountType.Savings },
                new[] { 1, 2, 10, 100, 150, -100 });
        }

        [Test]
        public void ApproveMessagerConfig3()
        {
            var st = new Stateprinter();
            st.Configuration.Add(new SbValueConverter());
            var messager = new NewRateEmailMessager();

            Config.Local.IncreaseRate = false;
            Config.Local.IncreaseRateFactor = 1.50m;

            CombinationApprovals.VerifyAllCombinations(
                (name, aType, rate) => messager.CreateMessage(name, aType, rate), x => st.PrintObject(x),
                new[] { "Vladimir Putin" },
                new[] { AccountType.Cheque, AccountType.Credit, AccountType.Savings },
                new[] { 1, 2, 10, 100, 150, -100 });
        }
    }
}