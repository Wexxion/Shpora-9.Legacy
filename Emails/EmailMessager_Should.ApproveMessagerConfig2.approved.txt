[Vladimir Putin,Cheque,1] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Vladimir Putin

We are sending you this message with respect to your chequing account

The interest rate at which you earn interest has changed to 0,75%.


Kind regards - your bank.

}
[Vladimir Putin,Cheque,2] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Vladimir Putin

We are sending you this message with respect to your chequing account

The interest rate at which you earn interest has changed to 1,50%.


Kind regards - your bank.

}
[Vladimir Putin,Cheque,10] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Vladimir Putin

We are sending you this message with respect to your chequing account

The interest rate at which you earn interest has changed to 7,50%.


Kind regards - your bank.

}
[Vladimir Putin,Cheque,100] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Vladimir Putin

We are sending you this message with respect to your chequing account

The interest rate at which you earn interest has changed to 75,00%.


Kind regards - your bank.

}
[Vladimir Putin,Cheque,150] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Vladimir Putin

We are sending you this message with respect to your chequing account

The interest rate at which you earn interest has changed to 112,50%.


Kind regards - your bank.

}
[Vladimir Putin,Cheque,-100] => new EmailMessage()
{
    Important = False
    Subject = "New rate!"
    Body = Dear Vladimir Putin

We are sending you this message with respect to your chequing account

The interest rate at which you earn interest has changed to -75,00%.


Kind regards - your bank.

}
[Vladimir Putin,Credit,1] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Vladimir Putin

We are sending you this message with respect to your credit card

The interest rate for which you will be charged for new purchases is now 0,75%


Kind regards - your bank.

}
[Vladimir Putin,Credit,2] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Vladimir Putin

We are sending you this message with respect to your credit card

The interest rate for which you will be charged for new purchases is now 1,50%


Kind regards - your bank.

}
[Vladimir Putin,Credit,10] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Vladimir Putin

We are sending you this message with respect to your credit card

The interest rate for which you will be charged for new purchases is now 7,50%


Kind regards - your bank.

}
[Vladimir Putin,Credit,100] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Vladimir Putin

We are sending you this message with respect to your credit card

The interest rate for which you will be charged for new purchases is now 75,00%


Kind regards - your bank.

}
[Vladimir Putin,Credit,150] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Vladimir Putin

We are sending you this message with respect to your credit card

The interest rate for which you will be charged for new purchases is now 112,50%


Kind regards - your bank.

}
[Vladimir Putin,Credit,-100] => new EmailMessage()
{
    Important = False
    Subject = "New rate!"
    Body = Dear Vladimir Putin

We are sending you this message with respect to your credit card

The interest rate for which you will be charged for new purchases is now -75,00%


Kind regards - your bank.

}
[Vladimir Putin,Savings,1] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Vladimir Putin

We are sending you this message with respect to your on line savings account

Your savings interest rate has changed to 0,75%


Kind regards - your bank.

}
[Vladimir Putin,Savings,2] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Vladimir Putin

We are sending you this message with respect to your on line savings account

Your savings interest rate has changed to 1,50%


Kind regards - your bank.

}
[Vladimir Putin,Savings,10] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Vladimir Putin

We are sending you this message with respect to your on line savings account

Your savings interest rate has changed to 7,50%


Kind regards - your bank.

}
[Vladimir Putin,Savings,100] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Vladimir Putin

We are sending you this message with respect to your on line savings account

Your savings interest rate has changed to 75,00%


Kind regards - your bank.

}
[Vladimir Putin,Savings,150] => new EmailMessage()
{
    Important = True
    Subject = "New rate!"
    Body = Dear Vladimir Putin

We are sending you this message with respect to your on line savings account

Your savings interest rate has changed to 112,50%


Kind regards - your bank.

}
[Vladimir Putin,Savings,-100] => new EmailMessage()
{
    Important = False
    Subject = "New rate!"
    Body = Dear Vladimir Putin

We are sending you this message with respect to your on line savings account

Your savings interest rate has changed to -75,00%


Kind regards - your bank.

}
